x-pack/plugins/xpack_main/server/lib/feature_registry/feature_registry.ts

```diff
@@ -37,7 +37,8 @@ export interface Feature {
   };
   catalogue?: string[];
   privileges: {
-    [key: string]: FeatureKibanaPrivileges;
+    all?: FeatureKibanaPrivileges;
```

Comment:
__question:__ what is the main reason we don't want to keep old signature here? To stress that we support only `all` and `read` privileges to TypeScript `registerFeature` consumers? It feels like at the rest of the places code is generalized enough and doesn't care about that fact.

If true, then maybe we can model this type a bit differently to avoid `if (!privilege)` within `Object.keys/values/entries` in most of the places, something along these lines:
```ts
type PrivilegesSet = Record<string, FeatureKibanaPrivileges>;
export interface Feature<TPrivileges extends Partial<PrivilegesSet> = PrivilegesSet> {
  ......
  privileges: TPrivileges;
  ......
}

export type FeatureWithAllOrReadPrivileges = Feature<{
  all?: FeatureKibanaPrivileges;
  read?: FeatureKibanaPrivileges;
}>;

// type FeatureWithoutPrivileges = Feature<never>;

.........
export class FeatureRegistry {
  public register(feature: FeatureWithAllOrReadPrivileges) {
    ........
    validateFeature(feature);
    ........
    this.features[feature.id] = feature as Feature;
  }
}

function validateFeature(feature: FeatureWithAllOrReadPrivileges) {
  .......
  Object.entries(feature.privileges).forEach(([privilegeId, privilegeDefinition]) => {
    if (!privilegeDefinition) {
      throw ......;
    }
  .......
}
```
What do you think @kobelb?

Reply:
> what is the main reason we don't want to keep old signature here? To stress that we support only `all` and `read` privileges to TypeScript `registerFeature` consumers? It feels like at the rest of the places code is generalized enough and doesn't care about that fact.

That's absolutely correct. Your suggestion works beautifully, and is much better than my original solution. Thanks so much @azasypkin!